cmd/agent/main.go

```diff
@@ -106,16 +111,26 @@ func (o *GrpcProxyAgentOptions) Validate() error {
 	if o.proxyServerPort <= 0 {
 		return fmt.Errorf("proxy server port %d must be greater than 0", o.proxyServerPort)
 	}
+	if o.serviceAccountTokenPath != "" {
+		if _, err := os.Stat(o.serviceAccountTokenPath); os.IsNotExist(err) {
+			return fmt.Errorf("error checking service account token path %s, got %v", o.serviceAccountTokenPath, err)
+		}
+	}
+
+	if o.agentCert == "" && o.agentKey == "" && o.serviceAccountTokenPath == "" {
+		return fmt.Errorf("agent must enable certificate based or token based authentication")
```

Comment:
I believe this should be enforced from the server side and not the client side. There are legitimate non production use cases for turning off client authentication.

Reply:
In addition if we allow this through I think we don't need the mock agent as the real agent can do what is necessary.

Reply:
We may want to add an enum setting to the proxy/main.go at this point for requiredAgentAuth? That way we can detect if the agent is meeting the minimum authentication requirements.

Reply:
I brought this up because an incorrectly configured konnectivity-agent would send a large number of denied requests to the konnectivity-server. (agent retries + multiple threads hitting LB for regional clusters + CrashLoop retries).

Since there are use cases for turning off client authentication, removing this check is fine, but we should still think of ways to limit the outgoing requests of an incorrectly configured agent.

Reply:
I removed this validation